pytroll_collectors/segments.py

```diff
@@ -635,20 +636,46 @@ def _generate_publish_service_name(self):
 
     def _setup_messaging(self):
         """Set up messaging."""
+        self._setup_listener()
+        self._setup_publisher()
+
+    def _setup_listener(self):
         self._subject = self._config['posttroll']['publish_topic']
         topics = self._config['posttroll'].get('topics')
         addresses = self._config['posttroll'].get('addresses')
+        services = self._config['posttroll'].get('services', "")
+        nameserver = self._config['posttroll'].get('nameservers')
+        if nameserver is not None:
+            if 'false' in nameserver:
+                nameserver = False
+            else:
+                nameserver = nameserver[0]
+        self._listener = ListenerContainer(
+            topics=topics,
+            addresses=addresses,
+            nameserver=nameserver,
+            services=services
+        )
+
+    def _setup_publisher(self):
+        publisher_config = self._collect_publisher_config()
+        self._publisher = create_publisher_from_dict_config(publisher_config)
+        self._publisher.start()
+
+    def _collect_publisher_config(self):
         publish_port = self._config['posttroll'].get('publish_port', 0)
         nameservers = self._config['posttroll'].get('nameservers', [])
-        services = self._config['posttroll'].get('services', "")
-        self._listener = ListenerContainer(topics=topics, addresses=addresses, services=services)
+        if nameservers is not None and 'false' in nameservers:
+            nameservers = False
```

Comment:
this is dublicated from above (geographic_gatherer), can it be refactored?

Reply:
Done.